nemo/collections/tts/models/fastpitch.py

```diff
@@ -511,45 +509,29 @@ def list_available_models(cls) -> 'List[PretrainedModelInfo]':
 
         return list_of_models
 
-    # Methods for model exportability
-    def _prepare_for_export(self, **kwargs):
-        super()._prepare_for_export(**kwargs)
-
-        # Define input_types and output_types as required by export()
-        self._input_types = {
-            "text": NeuralType(('B', 'T_text'), TokenIndex()),
-            "pitch": NeuralType(('B', 'T_text'), RegressionValuesType()),
-            "pace": NeuralType(('B', 'T_text'), optional=True),
-            "speaker": NeuralType(('B'), Index()),
-        }
-        self._output_types = {
-            "spect": NeuralType(('B', 'D', 'T_spec'), MelSpectrogramType()),
-            "num_frames": NeuralType(('B'), TokenDurationType()),
-            "durs_predicted": NeuralType(('B', 'T_text'), TokenDurationType()),
-            "log_durs_predicted": NeuralType(('B', 'T_text'), TokenLogDurationType()),
-            "pitch_predicted": NeuralType(('B', 'T_text'), RegressionValuesType()),
-        }
-
-    def _export_teardown(self):
-        self._input_types = self._output_types = None
-
     @property
     def disabled_deployment_input_names(self):
         """Implement this method to return a set of input names disabled for export"""
-        disabled_inputs = set()
+        disabled_inputs = set(["durs", "spec", "attn_prior", "mel_lens", "input_lens"])
         if self.fastpitch.speaker_emb is None:
             disabled_inputs.add("speaker")
         return disabled_inputs
 
+    @property
+    def disabled_deployment_output_names(self):
+        """Implement this method to return a set of output names disabled for export"""
+        disabled_outputs = set(["attn_soft", "attn_logprob", "attn_hard", "attn_hard_dur", "pitch"])
+        return disabled_outputs
+
     @property
     def input_types(self):
-        return self._input_types
+        return self.fastpitch.input_types
 
     @property
     def output_types(self):
-        return self._output_types
+        return self.fastpitch.output_types
 
-    def input_example(self, max_batch=1, max_dim=256):
+    def input_example(self, max_batch=1, max_dim=44):
         """
```

Comment:
If you'd prefer, I can make two separate PRs, but the rest of the changes are here because, if left as is, the script `scripts/export.py` will fail whenever one uses `--runtime-check` (see https://github.com/NVIDIA/NeMo/issues/4100#issuecomment-1116494057).